### PR TITLE
Update docs to be more general

### PR DESCRIPTION
**Pull request by BrendanChou**
- Less solo.js focused, includes docs for more repositories
- Removes unnecessary cover page
- Better dark theme
- Add short Python Client page
- Add search capability

### CODE REVIEW

File: docs/_coverpage.md [hunk 1]
```diff
@@ -1,13 +0,0 @@
-<p align="center"><img src="https://s3.amazonaws.com/dydx-assets/dydx_logo_white.svg" width="256" /></p>
```
Comment (AntonioJuliano) [inline docs/_coverpage.md]: I like the coverpage :( can we keep it?

We can add links to all the different projects on the coverpage which would be a good way to get to them

File: docs/_sidebar.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
-* [Home](/ "dYdX Solo Documentation")
-* [Protocol Overview](overview.md "dYdX Solo Documentation - Overview")
-* [Solo.js Usage](usage.md "dYdX Solo Documentation - Usage")
-* [HTTP API](api.md "dYdX Solo Documentation - API")
-* [Contracts](contracts.md "dYdX Solo Documentation - Contracts")
-* [Security](security.md "dYdX Solo Documentation - Security")
+- [Protocol](protocol.md "dYdX Documentation - Protocol")
+- [Solo.js](solojs.md "dYdX Documentation - Solo.js")
+- [HTTP API](api.md "dYdX Documentation - API")
+- [Python Client](python.md "dYdX Documentation - dYdX Python Client")
```
Comment (AntonioJuliano) [inline docs/_sidebar.md]: the title already has dYdX in it, don't need it again

File: docs/index.html [hunk 3]
```diff
@@ -2,32 +2,34 @@
...
 <head>
   <meta charset="UTF-8">
-  <title>@dydxprotocol/solo - Documentation</title>
+  <title>dYdX Documentation</title>
   <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1" />
-  <meta name="description" content="Ethereum Smart Contracts and TypeScript library used for the dYdX Solo-Margin Trading Protocol">
+  <meta name="description" content="Documentation for the dYdX Margin-Trading Protocol">
   <meta name="viewport" content="width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0">
-  <link rel="stylesheet" href="//unpkg.com/docsify/lib/themes/dark.css">
+  <link rel="stylesheet" href="//unpkg.com/docsify/lib/themes/vue.css">
```
Comment (AntonioJuliano) [inline docs/index.html]: I like it dark :(

File: docs/solojs.md [hunk 4]
```diff
@@ -1,4 +1,19 @@
-# Usage
+# Solo.js
```
Comment (AntonioJuliano) [inline docs/solojs.md]: Call this TypeScript Client to be consistent with "Python Client"

File: docs/_coverpage.md (hunk 1 above)
Comment (BrendanChou) [reply docs/_coverpage.md]: There is still a homepage, the coverpage is just gratuitous and adds an extra click.

File: docs/index.html (hunk 3 above)
Comment (BrendanChou) [reply docs/index.html]: changed to better dark theme

File: docs/_coverpage.md (hunk 1 above)
Comment (BrendanChou) [reply docs/_coverpage.md]: If there's anything you want to add to homepage, put it in README.md

File: docs/protocol.md [hunk 5]
```diff
@@ -1,4 +1,7 @@
-# Overview
+# Protocol
+
+- [GitHub](https://github.com/dydxprotocol/solo)
+
 Solo is an open protocol consisting of smart contracts that run on the Ethereum blockchain. Solo supports margin trading, borrowing, and lending. The Solo Typescript library makes it easy to interact with the smart contracts running on the blockchain.
```
Comment (AntonioJuliano) [inline docs/protocol.md]: The title says protocol, but this starts out "Solo.." which may be confusing for people who don't know what solo is

File: docs/typescript.md [hunk 6]
```diff
@@ -1,4 +1,10 @@
-# Usage
+# TypeScript Client
+
+TypeScript library for interacting with the dYdX V2 smart contracts
+
+- [GitHub](https://github.com/dydxprotocol/solo)
```
Comment (AntonioJuliano) [inline docs/typescript.md]: use the badges instead of these, you can pull them from the Github

File: docs/python.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
+# Python Client
+
+A python library for interacting with the Limit Order API.
+
+The library is currently tested against Python versions 2.7, 3.4, 3.5, and 3.6
+
+- [GitHub](https://github.com/dydxprotocol/dydx-python)
+- [PyPi](https://pypi.org/project/dydx-python)
```
Comment (AntonioJuliano) [inline docs/python.md]: Use badges here too (and add them to dydx-python if they don't exist there yet)

File: docs/python.md [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+## Install
+Install with `pip`:
+```
+pip install dydx-python
+```
+
+## Example Usage
+
+```python
```
Comment (AntonioJuliano) [inline docs/python.md]: When I run this locally, it doesn't syntax highlight this, I'm not sure why

Comment (BrendanChou) [reply docs/python.md]: yeah not sure either

File: docs/typescript.md [hunk 9]
```diff
@@ -1,4 +1,15 @@
-# Usage
+# TypeScript Client
+
+<br>
```
Comment (AntonioJuliano) [inline docs/typescript.md]: Can you both align these vertically and use the badges from the Github that auto pull the version and such? You should be able to simply copy from https://github.com/dydxprotocol/solo/blob/master/README.md

**Pull request merged**